### ISSUE TITLE
Sends Too Many Requests at Same Time

### ISSUE THREAD

**Issue by Jay54520** [maticnetwork/matic.js#210]
https://github.com/maticnetwork/matic.js/blob/74e13d6a2f4beeb71eda81c0a936038bba1cbc36/src/libs/ProofsUtil.ts#L243 sends more than 100 requests to RPC at same time while almost all the providers will discard your request, I have tried with Infura, maticvigil registed and ankr.
Improve: sends 10/other number of requests at same time, wait them finished and then next.

**Comment by ujjwalgupta94** [comment]
Good suggestion 👍 , i think its better to add configuration because limiting the call will slow the process. So its upto user to limit.

**Comment by Jay54520** [comment]
Why takes so long? Only need a few codes

```
  const asyncPool = require('tiny-async-pool')
  static async getReceiptProof(receipt, block, web3, receipts?) {
    const stateSyncTxHash = ethUtils.bufferToHex(ProofsUtil.getStateSyncTxHash(block))
    const receiptsTrie = new Trie()
    let txs = []
    if (!receipts) {
      for (const tx of block.transactions) {
        if (tx.hash === stateSyncTxHash) {
          // ignore if tx hash is bor state-sync tx
          return
        }
        txs.push(tx.hash)
      }
    }
    receipts = await asyncPool(10, txs, web3.eth.getTransactionReceipt)
```

**Comment by ujjwalgupta94** [comment]
@Jay54520 this is done but we are testing the implementation. Allow us some time please.

**Comment by ujjwalgupta94** [comment]
@Jay54520 this is implemented in v - 2.0.44. You can limit the concurrency when creating a client - 

```
import Matic from 'maticjs';

const maticPlasmaClient = new Matic({
    requestConcurrency: 5
})
```

Similarly can be done in POS client too.
